packages/jsii-pacmak/lib/targets/go.ts

```diff
@@ -78,27 +86,34 @@ export class Golang extends Target {
     // local build directory for this module. if
     // we do, add a "replace" directive to point to it instead of download from
     // the network.
-    const visit = (pkg: RootPackage) => {
-      for (const dep of pkg.packageDependencies) {
-        for (const baseDir of dirs) {
-          const moduleDir = tryFindLocalModule(baseDir, dep);
-          if (moduleDir) {
-            replace[dep.goModuleName] = moduleDir;
-          }
-        }
-
-        // recurse to transitive deps ("replace" is only considered at the top level go.mod)
-        visit(dep);
-      }
+    const visit = async (pkg: RootPackage): Promise<void> => {
+      return Promise.all(
+        pkg.packageDependencies.map((dep) =>
+          Promise.all(
+            dirs.map((baseDir) =>
+              tryFindLocalModule(baseDir, dep).then((moduleDir) => {
+                if (moduleDir) {
+                  replace[dep.goModuleName] = moduleDir;
+                }
+              }),
+            ),
+          ).then(() =>
+            // recurse to transitive deps ("replace" is only considered at the top level go.mod)
+            visit(dep),
+          ),
+        ),
+      ).then(() => void undefined);
     };
 
-    visit(this.goGenerator.rootPackage);
+    await visit(this.goGenerator.rootPackage);
 
     // write `local.go.mod`
 
     // read existing content
     const goMod = path.join(pkgDir, GOMOD_FILENAME);
-    const lines = (await fs.readFile(goMod, 'utf-8')).split('\n');
+    const lines = await fs
+      .readFile(goMod, 'utf8')
+      .then((text) => text.split('\n'));
```

Comment:
Does this handle errors properly?

Reply:
Yep - `.then` has no `catch` handler, so it'd simply use the original rejection.